core/src/main/scala/kafka/raft/RaftManager.scala

```diff
@@ -168,30 +166,20 @@ class KafkaRaftManager[T](
   }
 
   private def buildRaftClient(): KafkaRaftClient[T] = {
-    val quorumState = new QuorumState(
-      nodeId,
-      raftConfig.quorumVoterIds,
-      raftConfig.electionTimeoutMs,
-      raftConfig.fetchTimeoutMs,
-      new FileBasedStateStore(new File(dataDir, "quorum-state")),
-      time,
-      logContext,
-      new Random()
-    )
 
     val expirationTimer = new SystemTimer("raft-expiration-executor")
     val expirationService = new TimingWheelExpirationService(expirationTimer)
 
     new KafkaRaftClient(
-      raftConfig,
       recordSerde,
       netChannel,
       metadataLog,
-      quorumState,
+      new FileBasedStateStore(new File(dataDir, "quorum-state")),
```

Comment:
This is a really important file name and should be stored and documented in a static final field. How about defining this in `FileBasedStateStore` and changing the constructor to `FileBasedStateStore(Path dataDir)`?

If you agree, I can file an issue for this.

cc @aloknnikhil @hachikuji

Reply:
Makes sense to me.

Reply:
Wondering if it makes sense to introduce a config (or maybe just the final field) as part of #9916

Reply:
I don't think this should be configurable as it is part of the data (topic partition directory). It cannot conflict with other files in that directory and it cannot change between restart. KIP-595 documents this file and its use.

Reply:
Yea, that's fair. We can leave it for another PR.